thoth-api/migrations/v0.12.3/up.sql

```diff
@@ -4,3 +4,15 @@ ALTER TABLE series
 ALTER TABLE series
     ALTER COLUMN issn_digital DROP NOT NULL;
 
+ALTER TABLE work
+    ADD COLUMN withdrawn_date DATE;
+
+UPDATE work                              
+    SET withdrawn_date = updated_at    
+    WHERE (work_status = 'withdrawn-from-sale' 
+    OR work_status = 'out-of-print');
+
+ALTER TABLE work
+    ADD CONSTRAINT work_withdrawn_date_check CHECK
+        ((work_status = 'withdrawn-from-sale' OR work_status = 'out-of-print')
```

Comment:
the first condition should also include `AND withdrawn_date IS NOT NULL`

Reply:
Ah, I didn't realize that withdrawn_date was going to be required for out of print/withdrawn books ... I will change this in the migration, and also set the input to required in the app when work status is withdrawn or out of print.